device/WakeOnLan/controller.js

```diff
@@ -0,0 +1,155 @@
+﻿'use strict';
+
+const BluePromise = require('bluebird');
+const fs = require('fs')
+const readline = require('readline')
+const os = require('os');
+const net = require('net');
+const dgram = require('dgram');
+
+const PORT_NUMBER = 9;
+const DEFAULT_IPADDRESS = '255.255.255.255';
+
+let wolMap = "wol.map";
+
+/**
+ * Sets the path to the 'wol.map' file
+ */
+module.exports.setMapPath = (mapPath) => {
+    wolMap = mapPath;
+}
+
+/**
+ * Handles button presses for WOL
+ */
+module.exports.onButtonPressed = (deviceid, name) => {
+    console.log(`WOL button pressed for ${name}`);
+
+    parseAddr(name)
+        .then((addr) => {
+            return Promise.all([addr, getMacBuffer(addr.macAddress)])
+        })
+        .then((results) => {
+            let magic = Buffer.alloc(6, 0xff);
```

Comment:
why do you allocate a buffer with 6 bytes and the concatinate it 16 times. just init a buffer with the correct size. 

and you can also create a buffer object from a string: `Buffer.from(string, 'utf8');`